network/modules/UDP/socket_wrapper.py

```diff
@@ -0,0 +1,102 @@
+import socket
+
+
+class UdpSocket:
+    """
+    Wrapper for Python's socket module.
+    """
+
+    def __init__(self, socket_instance: socket.socket) -> None:
+        """
+        Parameters
+        ----------
+        instance: socket.socket
+            For initializing Socket with an existing socket object.
+        """
+        if socket_instance is None:
+            self.__socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        else:
+            self.__socket = socket_instance
+
+    def send_to(self, data: bytes, address: tuple) -> bool:
+        """
+        Sends data to specified address
+
+
+        Parameters
+        ----------
+        data: bytes
+        address: tuple
+
+
+        Returns
+        -------
+        bool: if data was transferred successfully
+
+        """
+        try:
+            self.__socket.sendto(data, address)
+        except socket.error as e:
+            print(f"Could not send data: {e}")
+            return False
+
+        return True
+
+    def recv_from(self, buf_size: int) -> "tuple[bool, bytes | None]":
```

Comment:
please edit the type hint. We want to return a tuple of length 4: (success, data, host, port)

Reply:
Additionally, unpack the data at the end once we've received everything